presto-main/src/main/java/io/prestosql/sql/planner/LocalDynamicFiltersCollector.java

```diff
@@ -79,33 +84,91 @@ public void collectDynamicFilterDomains(Map<DynamicFilterId, Domain> dynamicFilt
         // It is possible that some dynamic filters are collected in a different stage - and will not available here.
         // It is also possible that not all local dynamic filters are needed for this specific table scan.
-        List<ListenableFuture<TupleDomain<ColumnHandle>>> predicateFutures = symbolsMap.keySet().stream()
+        List<ListenableFuture<TupleDomain<ColumnHandle>>> predicateFutures = descriptorMap.keySet().stream()
                 .filter(futures.keySet()::contains)
                 .map(filterId -> {
                     // Probe-side columns that can be filtered with this dynamic filter resulting domain.
-                    List<ColumnHandle> probeColumns = symbolsMap.get(filterId).stream()
-                            .map(probeSymbol -> requireNonNull(columnsMap.get(probeSymbol), () -> format("Missing probe column for %s", probeSymbol)))
-                            .collect(toImmutableList());
+                    Map<ColumnHandle, ComparisonExpression.Operator> operatorsMap = descriptorMap.get(filterId).stream()
+                            .collect(toImmutableMap(
+                                    descriptor -> {
+                                        Symbol probeSymbol = Symbol.from(descriptor.getInput());
+                                        return requireNonNull(columnsMap.get(probeSymbol), () -> format("Missing probe column for %s", probeSymbol));
+                                    },
+                                    Descriptor::getOperator));
                     return Futures.transform(
                             requireNonNull(futures.get(filterId), () -> format("Missing dynamic filter %s", filterId)),
                             // Construct a probe-side predicate by duplicating the resulting domain over the corresponding columns.
                             domain -> TupleDomain.withColumnDomains(
-                                    probeColumns.stream()
+                                    operatorsMap.entrySet().stream()
                                             .collect(toImmutableMap(
-                                                    column -> column,
-                                                    column -> domain))),
+                                                    Map.Entry::getKey,
+                                                    entry -> applyComparison(entry.getValue(), domain)))),
                             directExecutor());
                 })
                 .collect(toImmutableList());
         return new TableSpecificDynamicFilter(predicateFutures);
     }
 
+    private static Domain applyComparison(ComparisonExpression.Operator operator, Domain domain)
+    {
+        switch (operator) {
+            case EQUAL:
+                return Domain.create(domain.getValues(), false);
+            case LESS_THAN: {
+                List<Range> ranges = domain.getValues().getRanges().getOrderedRanges();
```

Comment:
Do we need to get an ordered range here or can we use ` domain.getValues().getRanges().getSpan()` , then get low or high value from that ?